src/main/java/org/simulator/sbml/SBMLinterpreter.java

```diff
@@ -1220,16 +1195,20 @@ public void init(boolean renewTree, double defaultSpeciesValue, double defaultPa
         }
       }
     }
-    int sizeY = model.getCompartmentCount() + model.getSpeciesCount() + model.getParameterCount() + speciesReferencesInRateRules;
-    if (sizeY != Y.length) {
-      Y = new double[sizeY];
-      compartmentIndexes = new int[Y.length];
-      inConcentrationValues = new boolean[Y.length];
-      symbolIdentifiers = new String[Y.length];
-      conversionFactors = new double[Y.length];
-      Arrays.fill(conversionFactors, 1d);
-    }
 
+    int sizeY = model.getCompartmentCount() + model.getSpeciesCount() + model.getParameterCount() + speciesReferencesInRateRules;
+    Y = new double[sizeY];
+    oldY = new double[sizeY];
+    oldY2 = new double[sizeY];
```

Comment:
Maybe the name `oldY2` is not very intuitive. A comment would be helpful.

Reply:
@draeger, there is a comment for oldY2 at the time when it is first described. You can see it [here](https://github.com/hemilpanchiwala/SBSCL/blob/fern/src/main/java/org/simulator/sbml/EquationSystem.java#L352).

The comment is 

> Array for saving older Y values (when computing delayed values)

Reply:
Or better can we rename it to delayedY?

Reply:
Thanks for clarifying. I think that's good! It becomes clear when looking at the variable within an IDE rather than within a web browser.